lib/Doctrine/ORM/Query.php

```diff
@@ -489,16 +514,21 @@ public function useQueryCache($bool): self
     /**
      * Returns the cache driver used for query caching.
      *
+     * @deprecated
+     *
      * @return Cache|null The cache driver used for query caching or NULL, if
      * this Query does not use query caching.
      */
     public function getQueryCacheDriver(): ?Cache
     {
-        if ($this->queryCache) {
-            return $this->queryCache;
-        }
+        Deprecation::trigger(
+            'doctrine/orm',
+            'https://github.com/doctrine/orm/pull/9004',
+            '%s is deprecated and will be removed in Doctrine 3.0 without replacement.',
+            __METHOD__
+        );
 
-        $queryCache = $this->_em->getConfiguration()->getQueryCache();
+        $queryCache = $this->queryCache ?? $this->_em->getConfiguration()->getQueryCache();
 
         return $queryCache ? DoctrineProvider::wrap($queryCache) : null;
     }
```

Comment:
Let's take this as an occasion to add tests for this method? The added lines result in a drop in coverage.

Reply:
#9007